Edit README.md

shoki-0303 opened the pull request:
## WHAT
データベース設計の草案

## WHY
データベースを設計するため

shoki-0303 committed:
create_members_table_design

shoki-0303 committed:
create_messages_table_design

shoki-0303 committed:
create_groups_table_design

shoki-0303 committed:
create_users_table_design

shoki-0303 committed:
edit_README.md

no-iwamatsu-tech commented on README.md:
```diff
@@ -1,24 +1,51 @@
-* Configuration
+|Column|Type|Options|
+|------|----|-------|
+|body|text|null: false|
```
テキストなし、画像ありは許容したいので、DBにNOT NULL制約はつけずにRails側で判定するように実装しましょう！

no-iwamatsu-tech commented on README.md:
```diff
@@ -1,24 +1,51 @@
+|------|----|-------|
+|body|text|null: false|
+|image|text|----|
+|group_id|integer|foreign_key: true|
```
group単位でmessageを表示することになりますので、indexつけたいですね。

no-iwamatsu-tech commented on README.md:
```diff
@@ -1,24 +1,51 @@
-* Services (job queues, cache servers, search engines, etc.)
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, unique: true|
```
どちらでも良いですが、グループ検索などがあるわけではないので unique: true はない方のが良いと思います。
メンバーが増えグループが増えとどんどんサービスが拡大した場合に、汎用的な名前からどんどん使えなくなってしまいます。

no-iwamatsu-tech commented on README.md:
```diff
@@ -1,24 +1,51 @@
-* Deployment instructions
+### Association
+- has_many :messages
+- has_many :users, through::members
```
through: :membersと空白つけましょう。

no-iwamatsu-tech commented on README.md:
```diff
@@ -1,24 +1,51 @@
+
+### Association
+- has_many :messages
+- has_many :groups, through::members
```
ここもですね、through: と :members の間に空白欲しいです。

shoki-0303 committed:
modify_README.md

miyagit commented on README.md:
```diff
@@ -1,24 +1,51 @@
+- has_many :messages
+- has_many :users, through: :members
+- has_many :members
+- accept_nested_attributes_for :members
```
このプロパティ使わないです！

shoki-0303 committed:
delete_unnecessary_property

oyaoyaoyana commented on README.md:
```diff
@@ -1,24 +1,50 @@
-# README
+## membersテーブル
```
userやgroupなどのエンティティから順に記述をしましょう。

oyaoyaoyana commented on README.md:
```diff
@@ -1,24 +1,50 @@
+### Association
+- has_many :messages
+- has_many :users, through: :members
+- has_many :members
```
memberを経由して、userをとってきているので、
has_many :members
has_many :users, through: :members

oyaoyaoyana replied:
の順番が良いと思います。

oyaoyaoyana reviewed (commented):
コメントいたしました！

shoki-0303 committed:
remodify_README.md

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,50 @@
-application up and running.
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false|
```
null: falseは外しても良いと思います。
理由は、groupが増えていくと後々凡庸な名前から登録出来なくなるからです。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,50 @@
-* Configuration
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, unique: true|
```
後半でnameを検索する機能を実装するのでindexオプションを貼りましょう。

MrUrakou reviewed (commented):
idはreference型を使用しましょう。
integer型で定義してその後indexを貼ってもよいですが、記述量も増えるため定義時にreference型かつforeign_key :trueかつnull: falseをつけたほうが良いです！
参考url
https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

shoki-0303 committed:
modify_option_README.md

In groups_table , delete null: false at name column.

In users_table, add index: true  at name column.

shoki-0303 committed:
edit_README.md

setup references
